clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -527,17 +565,28 @@ PointerVariableConstraint::mkString(EnvironmentMap &E,
     if (!ForItype && BaseType == "void")
       K = Atom::A_Wild;
 
+    if (PrevArr && K != Atom::A_Arr && !EmittedName) {
+      EmittedName = true;
+      addArrayAnnotations(CheckedArrs, EndStrs);
+      EndStrs.push_front(" " + getName());
+    }
+    PrevArr = ((K == Atom::A_Arr || K == Atom::A_NTArr)
+               && ArrPresent
```

Comment:
What does `ArrPresent` mean and where does it come from? It's not in your list of locals, above.

Reply:
ArrPresent is a field in PointerVariableCosntraint object, determining if the PVConstraint refers to an Array